addons/project_hr_expense/models/project_project.py

```diff
@@ -22,7 +22,7 @@ def _get_expense_action(self, domain=None, expense_ids=None):
             'context': {'project_id': self.id},
             'domain': domain or [('id', 'in', expense_ids)],
         })
-        if len(expense_ids) == 1:
+        if not self.env.context.get('from_embedded_action', False) and len(expense_ids) == 1:
```

Comment:
By default, `get` from a dict return `None` so, it is not really needed to set `False` if the key is not found since `None` will be evaluated as `False`, here. 🙂 

(no need to change, of course, it is just for your information)